doc/contributing/collaborator-guide.md

```diff
@@ -378,6 +380,20 @@ an effort to determine the potential impact of the change in the ecosystem. Use
 If a change will cause ecosystem breakage, then it is semver-major. Consider
 providing a Public API in such cases.
 
+### Breaking changes to console APIs
+
+Console APIs are a debugging tool, and the format of their output should generally not
+considered stable.
+However due to the nature of the Node.js ecosystem, users rely on stability of
+the output of some of these APIs for snapshot testing, parsers, etc...
+To avoid breaking changes, some behaviors should be considered stable.
+The following changes in the output are to be considered breaking changes:
+
+* `console.table` (padding, alignment, etc...).
+* changing the color of all console methods.
```

Comment:
I feel like this is less about the color (visual appearance when connected to a TTY) and more about the contract of outputting raw strings unmodified to the output stream when piped (without adding additional control sequences).

The visual color of the output can change easily as `util.inspect()` is evolved and doesn't matter for interactive use, but a lot of users rely on `console.error(str)` or `console.log(str)` to output `str` untouched to `process.stderr`/`process.stdout`, e.g. for logging JSON in production.

Reply:
to be fair `process.stdout`/`stderr` `.write()` has always been available to use for these needs. The pain is around correcting all the code that needs this changed. One might argue it to be superior to make a NEW set of console or other APIs that could gain features like this and maybe steadily enhance those features over time. 

I have to assume that this was the whole point of the console interface from the beginning... After all, it has slick stuff like `console.table()` that everyone knows not to rely on its implementation specific properties.